src/App.js

```diff
@@ -1,29 +1,15 @@
 import React, { Component } from 'react';
-import { IndexRoute, Router, Route, browserHistory } from 'react-router';
-import { render } from 'react-dom';
 
 import './App.css';
 
-import Home from './components/Home/index.jsx';
-import About from './components/About/index.jsx';
-
 class App extends Component {
   render() {
     return (
       <div className="App">
-        <Home />
+        {this.props.children}
```

Comment:
this is the real problem -- you were rendering the App component, which always renders `<Home />` regardless of route